source/beamline/BioXAS/BioXASSideBeamStatusControl.cpp

```diff
@@ -0,0 +1,185 @@
+#include "BioXASSideBeamStatusControl.h"
+#include "actions3/AMActionSupport.h"
+
+BioXASSideBeamStatusControl::BioXASSideBeamStatusControl(const QString &name, QObject *parent) :
+	BioXASBeamStatusControl(name, parent)
+{
+	// Initialize inherited variables.
+
+	setContextKnownDescription("SideBeamStatus");
+
+	// Initialize local variables.
+
+	frontEndStatus_ = 0;
+	poeBeamStatus_ = 0;
+	soeBeamStatus_ = 0;
+
+	// Current settings.
+
+	updateStates();
+}
+
+BioXASSideBeamStatusControl::~BioXASSideBeamStatusControl()
+{
+
+}
+
+bool BioXASSideBeamStatusControl::shouldStop() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->shouldStop();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canMeasure() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canMeasure();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canMove() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canMove();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::canStop() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->canStop();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::isOn() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->isOn();
+
+	return result;
+}
+
+bool BioXASSideBeamStatusControl::isOff() const
+{
+	bool result = false;
+
+	if (soeBeamStatus_)
+		result = soeBeamStatus_->isOff();
+
+	return result;
+}
+
+void BioXASSideBeamStatusControl::setFrontEndBeamStatus(BioXASFrontEndBeamStatusControl *newControl)
+{
+	if (frontEndStatus_ != newControl) {
+
+		if (poeBeamStatus_)
```

Comment:
The POE control depends on the front-end control. What I mean to do here is to update the  front-end control in the POE control when a new front-end control is set here.